src/hooks/useTableFilters.ts

```diff
@@ -0,0 +1,129 @@
+import { useState } from "react";
+import { useLocation, useHistory } from "react-router-dom";
+import { parseQueryString } from "utils";
+import { updateUrlQueryParam } from "utils/url";
+
+type InputEvent = React.ChangeEvent<HTMLInputElement>;
+
+interface Params<SearchParam> {
+  urlSearchParam: SearchParam; // the name of url param
+  sendAnalyticsEvent: (filterBy: string) => void; // callback function to send analytics event
+}
+
+type UseInputFilterReturn = [
+  string, // url param value
+  (e: InputEvent) => void, // onChange handler
+  () => void, // update url param
+  () => void // reset url param
+];
+
+// USE FOR FILTERS BUILT INTO TABLE COLUMN HEADERS
+export const useTableInputFilter = <SearchParam extends string>({
+  urlSearchParam,
+  sendAnalyticsEvent = () => undefined,
+}: Params<SearchParam>): UseInputFilterReturn => {
+  const { pathname, search } = useLocation();
+  const { replace } = useHistory();
+
+  const urlParams = parseQueryString(search);
+
+  const inputValueFromUrl = (urlParams[urlSearchParam] || "").toString();
+
+  const [value, setValue] = useState(inputValueFromUrl);
+
+  const onChange = (e: InputEvent): void => {
+    setValue(e.target.value);
+  };
+
+  const updateParams = () => {
+    updateUrlQueryParam(
+      urlSearchParam,
+      value,
+      search,
+      replace,
+      pathname,
+      sendAnalyticsEvent,
+      true
+    );
+
+    sendAnalyticsEvent(urlSearchParam);
+  };
+
+  const resetQueryParam = () => {
+    setValue("");
```

Comment:
I think we should be setting the value to the original query param values that are reflected in the filter component before any selections or deselections were made. 
It seems like `setValue("")` will deselect all of the checkboxes.

Reply:
the line you're commenting on is for the input filters, not checkboxes. 

reset = remove filter = setting it to the default value, which in this case is an empty string